### DISCUSSION TITLE
Test Case Library

### DISCUSSION THREAD

**Discussion by hkolbeck-streem** [kdl-org/kdl#78] [General]
As part of the development of [kdl4j](https://github.com/hkolbeck/kdl4j) I wrote up a fairly large set of test cases with the intent that they be usable by other implementations. At present there are about 160, and I hope they are useful to any other parsing library authors. Each test case consists of an input file and possibly an expected output file. If an output file exists, it means parsing should succeed and result in a document identical to the output file. In no output file exists, it means that parsing should fail.

Because there is currently no standard for indentation, the presence of semicolons, etc, I had to pick one. If a standard is ever chosen I will update. I would very much welcome PRs adding additional cases, especially tricky ones that should fail to parse.

They can be found here: https://github.com/hkolbeck/kdl4j/tree/trunk/src/test/resources, and the README there explains their usage more completely. The Java code I use to process them can be found here: https://github.com/hkolbeck/kdl4j/blob/trunk/src/test/java/dev/hbeck/kdl/TestRoundTrip.java

**Comment by hkolbeck-streem** [comment]
I've moved these test cases to a CC0 public domain license: [License](https://github.com/hkolbeck/kdl4j/blob/trunk/LICENSE.md)

Attribution is still appreciated, but not required.

**Comment by scotttrinh** [comment]
This is also helpful for generating some examples for the website, too, thanks for this!

**Comment by zkat** [comment]
I've converted this into an issue: #115 because I want this resolved by 1.0.0
